### PR TITLE
fix: don't leak socket if client is destroyed while connecting

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by mcollina** [comment]
CI is failing :( and this might need a test.

**Comment by ronag** [comment]
@mcollina PTAL

**Review by mcollina — APPROVED** [review]
lgtm

File: types/connector.d.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ declare namespace buildConnector {
     port: string
     servername?: string
     localAddress?: string | null
-    httpSocket?: Socket
+    httpSocket?: Socket,
```
Comment (ronag) [inline types/connector.d.ts:24]:
```suggestion
    httpSocket?: Socket
```

**Pull request merged**